bluemira/geometry/parameterisations.py

```diff
@@ -1189,43 +1336,633 @@ def create_shape(self, label=""):
         shape: BluemiraWire
             CAD Wire of the geometry
         """
-        x1, x2, x3, z1_frac, z2, z3, r = self.variables.values
-        z1 = z1_frac * z2
-        p1 = [x3 - r, 0, z3]
-        p2 = [x2, 0, z3]
-        p3 = [x2, 0, z2]
-        p4 = [x1, 0, z1]
-        p5 = [x1, 0, -z1]
-        p6 = [x2, 0, -z2]
-        p7 = [x2, 0, -z3]
-        p8 = [x3 - r, 0, -z3]
-        c1 = [x3 - r, 0, -z3 + r]
-        p9 = [x3, 0, -z3 + r]
-        p10 = [x3, 0, z3 - r]
-        c2 = [x3 - r, 0, z3 - r]
+        x1, x2, x3, z1, z2, ri, ro = self.variables.values
+        p5 = [x3, 0, z2 - ro]
+        p6 = [x3, 0, -z2 + ro]
+
         axis = [0, -1, 0]
+        wires = []
+        inb_leg = PictureFrameTools._make_tapered_inner_leg(
+            axis, x1, x2, z1, z2 - ri, -z2 + ri
+        )
+        wires.append(inb_leg)
 
-        wires = [make_polygon([p1, p2, p3, p4, p5, p6, p7, p8], label="inner_limb")]
+        top_leg = PictureFrameTools._make_flat_leg(
+            axis,
+            x2,
+            x3,
+            z2,
+            ri,
+            ro,
+            flip=False,
+        )
+        wires.append(top_leg)
 
-        if r != 0.0:
-            wires.append(
-                make_circle(
-                    r,
-                    c1,
-                    start_angle=270,
-                    end_angle=360,
-                    axis=axis,
-                    label="lower_corner",
-                )
-            )
+        wires.append(make_polygon([p5, p6], label="outer_limb"))
 
-        wires.append(make_polygon([p9, p10], label="outer_limb"))
+        bot_leg = PictureFrameTools._make_flat_leg(
+            axis,
+            x2,
+            x3,
+            -z2,
+            ri,
+            ro,
+            flip=True,
+        )
+        wires.append(bot_leg)
+        return BluemiraWire(wires, label=label)
 
-        if r != 0.0:
-            wires.append(
-                make_circle(
-                    r, c2, start_angle=0, end_angle=90, axis=axis, label="upper_corner"
-                )
-            )
+
+class FullDomeFlatInnerCurvedPictureFrame(GeometryParameterisation):
+    """
+    Curved (Superconducting) picture-frame geometry parameterisation, with
+    top and bottoms domed
```

Comment:
Argument description

Reply:
So I just followed the same format as all the other parameterisations

Reply:
Bluemira is using the Sphinx doc convention and here is an example of the a class doc in sphinx

![image](https://user-images.githubusercontent.com/55831357/154121617-18064052-4f51-4f62-a5a5-9c1fb510590b.png)

Reply:
The docstring convention we use is described in CONTRIBUTING.md: https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/develop/CONTRIBUTING.md.